R/next_cursor.R

```diff
@@ -92,182 +92,17 @@ next_cursor.response <- function(x) {
-        x <- x[[1]][["id"]]
-      } else if (has_name_(x[[1]], "ids")) {
-        x <- x[[1]][["ids"]]
-      }
-    } else if (has_name_(x, "status_id")) {
-      x <- x[["status_id"]]
-    } else if (has_name_(x, "user_id")) {
-      x <- x[["user_id"]]
-    }
-  }
-  ##return_last(x) + adj
-  if (!is.recursive(x)) {
-    id_minus_one(return_last(x))
-  } else {
-    NULL
-  }
-}
-
-
 #' @rdname next_cursor
 #' @param .x id
 #' @export
-max_id <- function(.x) UseMethod("max_id")
-
-#' @export
-max_id.default <- function(.x) {
-  if (inherits(.x, "response")) {
-    .x <- from_js(.x)
-  } else {
-    return(NULL)
-  }
-  max_id(.x)
-}
-
-#' @export
-max_id.character <- function(.x) {
-  .x[length(.x)]
-}
-
-#' @export
-max_id.NULL <- function(.x) return(NULL)
-
-#' @export
-max_id.numeric <- function(.x) {
-  sp <- getOption("scipen")
-  on.exit(options(scipen = sp), add = TRUE)
-  options(scipen = 15)
-  .x <- as.character(.x)
-  max_id(.x)
-}
-
-#' @export
-max_id.integer <- function(.x) {
-  sp <- getOption("scipen")
-  on.exit(options(scipen = sp), add = TRUE)
-  options(scipen = 15)
-  .x <- as.character(.x)
-  max_id(.x)
-}
-
-
-#' @export
-max_id.factor <- function(.x) {
-  .x <- as.character(.x)
-  max_id(.x)
-}
-
-
-#' @export
-max_id.data.frame <- function(.x) {
-  if (has_name_(attributes(.x), "max_id_str")) return(attr(.x, "max_id_str"))
-  if (has_name_(attributes(.x), "max_id")) return(attr(.x, "max_id"))
-  idvar <- c("status_id", "id_str", "id")
-  if (nrow(.x) > 0L && any(idvar %in% names(.x))) {
-    idvar <- idvar[idvar %in% names(.x)][1]
-    .x <- .x[[idvar]]
-  } else if (nrow(.x) > 0L && "user_id" %in% names(.x) &&
-      any(c("description", "profile_image_url", "friends_count") %in% names(.x))) {
-    stop("Failed to find status ID variable. You may have specified users data by mistake.")
-  } else {
-    .x <- NULL
-  }
-  max_id(.x)
-}
-
-#' @export
-max_id.list <- function(.x) {
-  if (has_name_(.x, "max_id_str")) return(.x[["max_id_str"]])
-  if (has_name_(.x, "max_id")) return(.x[["max_id"]])
-  while (is_emptylist(.x)) {
-    .x <- .x[[1]]
-  }
-  if (is.null(names(.x)) &&
-        any(vapply(.x, function(x) isTRUE("statuses" %in% names(x)),
-                   logical(1)))) {
-    .x <- lapply(.x, "[[", "statuses")
-  }
-  if (is.null(names(.x)) && any(vapply(.x, is.recursive, logical(1)))) {
-    .x <- .x[lengths(.x) > 0L & vapply(.x, is.recursive, logical(1))]
-    .x <- .x[[length(.x)]]
-  }
-  if (isTRUE("statuses" %in% names(.x))) {
-    .x <- .x[["statuses"]]
-  }
-  if (is.null(.x) || length(.x) == 0L) return(NULL)
-  .x <- tryCatch(
-    as.data.frame(.x[!vapply(.x, is.recursive, logical(1))],
-                  row.names = NULL, stringsAsFactors = FALSE),
-    error = function(e) return(NULL)
-  )
-  max_id(.x)
-}
-
-#' @export
-max_id.response <- function(.x) {
-  .x <- from_js(.x)
-  max_id(.x)
+max_id <- function(.x) {
+  stop("max_id() is deprecated; rtweet now handles pagination for you")
```

Comment:
This doesn't make use of `lifecycle::deprecate_warn`, not sure how it was used previously so not sure if this error each time is more user friendly than a warning.

Reply:
I think this was mostly for internal use; we can bring it back later if we discover folks need it.